kafka/src/main/java/io/specmesh/kafka/Provisioner.java

```diff
@@ -127,4 +133,82 @@ static ParsedSchema getSchema(final String topicName, final String schemaRef, fi
         }
         throw new RuntimeException("Failed to handle topic:" + topicName + " schema: " + path);
     }
+
+    /**
+     * Provision Topics, ACLS and schemas
+     *
+     * @param apiSpec
+     *            given spec
+     * @param schemaResources
+     *            schema path
+     * @param adminClient
+     *            kafka admin client
+     * @param schemaRegistryClient
+     *            sr client
+     * @throws ExecutionException
+     *             when cant provision topics
+     * @throws InterruptedException
+     *             process interrupted
+     * @throws TimeoutException
+     *             stalled
+     */
+    public static void provision(final KafkaApiSpec apiSpec, final String schemaResources,
+            final AdminClient adminClient, final SchemaRegistryClient schemaRegistryClient)
+            throws ExecutionException, InterruptedException, TimeoutException {
```

Comment:
This class throws a lot of checked exceptions from its public API.  

Are we expecting users to handle each type differently?  If we're not, then we should just wrap the checked exception in our own (inner) `ProvisioningException` runtime exception.

Oh, and extract the inner cause from `ExecutionException`.  `ExecutionException` is an implementation detail, i.e. you're using futures or similar.  Extract the cause and wrap.

Reply:
yeah - I wrapped them to an inner